specifications/wallet/spec-light-mode.md

```diff
@@ -0,0 +1,143 @@
+  [mithril]: https://iohk.io/en/research/library/papers/mithrilstake-based-threshold-multisignatures/
+
+# Motivation
+
+## Background
+
+As the Cardano blockchain grows in size, retrieving and verifying blocks from the consensus network becomes increasingly time consuming. By making a "trust versus speed" trade-off, we can significantly decrease synchronization times.
+
+## User-visible benefits
+
+* Allow users to operate their wallet without needing to wait for the node to download and validate the entire blockchain.
+* Allow users to run cardano-wallet and Daedalus on systems with significantly less than 8GB of memory and less than 10GB of disk space.
+* Allow users to control where they sit on the trust vs convenience spectrum, depending upon their current circumstances.
+
+## Technical benefits
+
+* *Speed*. With light-mode, we expect synchronisation times of < 1 minute for a wallet with 1'000 transactions. In contrast, synchronisation of an empty wallet currently takes ~ 50 minutes by connecting to a node.
+* *Compatiblity*. Light-mode is intended to preserve the API presented to downstream software such as [Daedalus][], only *minimal* changes are expected.
+* *Optionality*. A wallet that was started in light-mode can be restarted in full mode without resynchronization, and vice versa. (MVP: no support yet for changing the mode dynamically while the wallet is running.)
+
+## Limitations
+
+* *Trust*. The external data source does *not* provide the same level of protection against *omission* of transactions as the Proof-of-Stake consensus protocol.
+* *Privacy*. In addition to the reduction of trust in the blockchain data, we now also reveal private information about addresses belonging to a single wallet.
+* *Address schemes*. Only *Shelley* wallets with *sequential address discovery* can be supported.
+
+However, these limitations are shared by all existing light wallets. In fact, some light wallets only provide *single-address* wallets, which is an additional reduction of privacy.
+
+# Specification
+
+## Overview
+
+The implementation of light-mode is based on an efficient query `Address -> Transactions` which the blockchain data source provides. This query is useful to wallets that use *sequential address discovery* (Shelley wallets). These wallets work with a sequence of potential addresses `addr_0`, `addr_1`, `addr_2`, …. For each integer `n`, there is a deterministic procedure for generating the corresponding address `addr_n`. The wallet generates the first few addresses in this sequence and uses the above query to retrieve the corresponding transactions. When no transactions are found for `g` ("address gap") many consecutive addresses, the procedure stops — as the wallet never puts addresses with higher indices on the chain. In other words, this iterative loop yields all transactions that belong to the wallet.
+
+This procedure is implemented and demonstrated in the `light-mode-test` prototype.
+
+## Functionality
+
```

Comment:
And here is another gratuitous picture opportunity.
Could we please have two block diagrams - full-node and light-mode?
Like a cut-down version of the diagram here: https://input-output-hk.github.io/cardano-wallet/developers/Architecture
In the "full-node" version, wallet links to node in 4 ways: chainsync, localblockfetch, localstatequery, localtxsubmission.
In the "light-mode" version, wallet requests a list of transactions by address from the block service.

Reply:
Ok!

Note: We do not use `BlockFetch`, but `ChainSync` to receive the blocks from the node. This is line with the network specification, which recommends that

> The (full)-node to client protocol uses an instance of the chain synchronisation protocol that transfers full blocks, while the node-to-node instance only transfers block headers. In the node-to-node case, the block fetch protocol (Section 3.8) is used to transfer full blocks.